ex2_hal/adcs/hardware_interface/include/adcs.h

```diff
@@ -67,7 +67,9 @@ typedef struct __attribute__((packed)) {
     float Mag_Field_Vector_X;
     float Mag_Field_Vector_Y;
     float Mag_Field_Vector_Z;
-    uint16_t Comm_Status;
+    uint16_t TC_num;
+    uint16_t TM_num;
+    uint8_t CommsStat_flags[6];
```

Comment:
It's nice not to have literals around, since it's easy to get them wrong and hard to change them (only particular instances of 4, for example). I recognize that this stuff has an insane number of magic numbers, but still...